File: _c++/symulacja_organizmow/Organizmy/Zwierzeta/Zolw.cpp
//
//  Zolw.cpp
//  WirtualnySwiat
//
//  Created by Michal Tarnacki on 10/04/2022.
//

#include "Zolw.hpp"

Zolw::Zolw(Swiat &swiat):Zwierze(swiat){
    SetInicjatywa();
    SetSila();
};
void Zolw::Akcja(){
    int x=rand()%100;
    if(x>=75)
        Zwierze::Akcja();
}
void Zolw::Kolizja(Organizm& organizm){
    if(organizm.GetSila()<5&&!dynamic_cast<Zolw*>(&organizm)){
        organizm.SetPolozenie(organizm.GetPoprzedniePolozenie());
        std::cout << "Organizm: " << organizm.GetPolozenie() << " jest za slaby zeby walczyc z zolwiem, wraca na poprzenia pozycje: " << organizm.GetPolozenie() << std::endl;
    }
    else
        Zwierze::Kolizja(organizm);
}
char Zolw::Rysowanie() const{
    return 'Z';
}
void Zolw::SetInicjatywa(){
    this->inicjatywa=1;
}
void Zolw::SetSila(){
    this->sila=2;
}
Organizm* Zolw::Urodz(){
    return new Zolw(this->swiat);
}
bool Zolw::MozeBycPartnerem(Organizm& organizm){
    if(dynamic_cast<Zolw*>(&organizm))
       return true;
    else
       return false;
}

File: _java/symulacja_organizmow/src/Symulacja/Zwierzeta/Zolw.java
package Symulacja.Zwierzeta;
import Symulacja.*;

import java.awt.*;

public class Zolw extends Zwierze {
    protected void SetInicjatywa(){
        this.inicjatywa=1;
    }
    protected void SetSila(){
        this.sila=2;
    }
    public Zolw(Swiat swiat){
        super(swiat);
    }
    public char Rysowanie(){
        return 'Z';
    }
    public Organizm Urodz() {
        return new Zolw(this.swiat);
    }
    public boolean MozeBycPartnerem(Organizm organizm){
        if(organizm instanceof Zolw)
            return true;
        else
            return false;
    }
    public void Akcja(){
        int x=(int)(Math.random()*100);
        if(x>=75)
            super.Akcja();
    }
    public void Kolizja(Organizm organizm){
        if(organizm.GetSila()<5&&!(organizm instanceof Zolw)){
            organizm.SetPolozenie(organizm.GetPoprzedniePolozenie());
            swiat.zdarzenia.DodajString( "Organizm: " + organizm.GetPolozenie() + " jest za slaby zeby walczyc z zolwiem, wraca na poprzenia pozycje: " + organizm.GetPolozenie());
        }
    else
        super.Kolizja(organizm);

    }
    @Override
    public Color GetColor(){
        return new Color(0,100,100);
    };
}
